src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaHealthIndicator.java

```diff
@@ -0,0 +1,92 @@
+package de.tum.in.www1.artemis.service.connectors.athena;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+
+/**
+ * Service determining the health of the Athena service and its assessment modules.
+ */
+@Component
+@Profile("athena")
+public class AthenaHealthIndicator implements HealthIndicator {
+
+    private static final String GREEN_CIRCLE = "\uD83D\uDFE2"; // unicode green circle
+
+    private static final String RED_CIRCLE = "\uD83D\uDD34"; // unicode red circle
+
+    private final RestTemplate shortTimeoutRestTemplate;
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    public AthenaHealthIndicator(@Qualifier("shortTimeoutAthenaRestTemplate") RestTemplate shortTimeoutRestTemplate) {
+        this.shortTimeoutRestTemplate = shortTimeoutRestTemplate;
+    }
+
+    private record AthenaModuleHealth(String type, boolean healthy, String url) {
+    }
+
+    private static String moduleHealthToString(AthenaModuleHealth moduleHealth) {
+        var healthString = moduleHealth.healthy ? GREEN_CIRCLE : RED_CIRCLE;
+        healthString += " " + moduleHealth.url + " (" + moduleHealth.type + ")";
```

Comment:
I feel like this is more readable and the performance gains are negligible (because this is not within a loop). Besides, the Java compiler performs such optimizations by itself anyways, apparently: https://stackoverflow.com/a/11942386/4306257

Reply:
I agree, outside of loops StringBuilders are not needed, Java optimizes that internally.